runtime/compiler/runtime/Listener.cpp

```diff
@@ -45,10 +45,34 @@
 #include "runtime/Listener.hpp"
 
 static bool
-handleOpenSSLConnectionError(int connfd, SSL *&ssl, BIO *&bio, const char *errMsg)
+handleOpenSSLConnectionError(int connfd, SSL *&ssl, BIO *&bio, const char *errMsg, int ret, TR::CompilationInfo *compInfo)
    {
+   int errnoCopy = errno;
+   int sslError = SSL_ERROR_NONE;
+   unsigned long errorCode = 0;
+   int errorReason = 0;
+   char errorString[256] = {0};
+
+   // Filter out SSL_R_UNEXPECTED_EOF_WHILE_READING errors that were introduced in SSL version 3
+   // These can appear if Kubernetes readiness/liveness probes are used on the TCP encrypted channel
+   if (ret <= 0)
+      {
+      sslError = (*OSSL_get_error)(ssl, ret);
+      // Get earliest error code from the thread's error queue without modifying it.
+      errorCode = OERR_peek_error();
+      errorReason = ERR_GET_REASON(errorCode);
+      OERR_error_string_n(errorCode, errorString, sizeof(errorString));
+      if (sslError == SSL_ERROR_SSL && errorReason == SSL_R_UNEXPECTED_EOF_WHILE_READING)
+         {
+         // Take this error out of the queue so that it doesn't get printed again
```

Comment:
When you say "printed again", when was it printed before?

Reply:
I deleted "again". 
The error will get printed in `(*OERR_print_errors_fp)(stderr);` if we don't take it out of the queue.